internal/pkg/remote/client.go

```diff
@@ -220,13 +220,30 @@ func (c *Client) Send(
 			c.log.With("error", err).Debugf(msg)
 			continue
 		}
+		c.checkApiVersionHeaders(reqID, resp)
 
 		return resp, nil
 	}
 
 	return nil, fmt.Errorf("all hosts failed: %w", multiErr)
 }
 
+func (c *Client) checkApiVersionHeaders(reqID string, resp *http.Response) {
+	const elasticApiVersionHeaderKey = "Elastic-Api-Version"
+	const warningHeaderKey = "Warning"
+
+	warning := resp.Header.Get(warningHeaderKey)
+	if warning != "" {
+		c.log.With("http.request.id", reqID).Warnf("warning in fleet response: %q", warning)
+	}
+
+	if downgradeVersion := resp.Header.Get(elasticApiVersionHeaderKey); resp.StatusCode == http.StatusBadRequest && downgradeVersion != "" {
```

Comment:
Is there any relationship between the warning header and the API version header? Is it possible to have one set but not the other? In the context of version negotiation, is there some specific value of the warning header that should make Agent look at the value of the API version header?

Reply:
According to the related issue:
- we have `Elastic-Api-Version` ina  response with an http status code `400` when we are using an api version that is too new
- we get a warning if the Api version is too old but fleet server will attempt to serve the request anyways (it won't immediately answer with a `Bad Request` code

I am not sure that we have overlap (like if we get a warning we could also get a significant value of `Elastic-Api-Version` header) but maybe @joshdover  can answer that ;)

Reply:
Reading through the related issue, it doesn't look like there's any strong connection between the content of the warning header and the API version header. So I think the logic here is good as-is.